helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -117,9 +121,20 @@ public HelixDataAccessor getDataAccssor(String clusterName) {
     }
   }
 
+  public ZkBaseDataAccessor<ZNRecord> getZkBaseDataAccessor(ZkSerializer serializer) {
+    synchronized (_zkBaseDataAccessorBySerializer) {
+      if (!_zkBaseDataAccessorBySerializer.containsKey(serializer)) {
+        ZkBaseDataAccessor<ZNRecord> baseDataAccessor = new ZkBaseDataAccessor<>(_zkAddr, serializer);
+        _zkBaseDataAccessorBySerializer.put(serializer, baseDataAccessor);
+      }
+    }
+    return _zkBaseDataAccessorBySerializer.get(serializer);
```

Comment:
Should we close the BaseDataAccessor somewhere (instead of keeping it  forever until the server being shutdown)?

Reply:
*revise* my comment elsewhere, the `close` method in the class will close the connection when server is shut down. I don't see the right time in the rest server when we should close the connection: if closing it after request finished, the next time, it needs to establish a new connection as well.

Reply:
Ping - see my comment about 20K different serializers?

Reply:
no longer valid